### ISSUE TITLE
macos get two disk but my mac only have one disk

### ISSUE THREAD

**Issue by fireinrain** [oshi/oshi#2303]
package version   <oshiVersion>5.3.6</oshiVersion>

<img width="1158" alt="image" src="https://user-images.githubusercontent.com/14249262/216271099-75cf0697-65da-4fd1-9fb0-58a9b8332198.png">

my mac run in macos catalina, and hardware is 500GB with one storage disk.

**Comment by dbwiddis** [comment]
This is normal since Catalina.   See https://discussions.apple.com/thread/250820445 or open your Disk Utility to explore on your own.

**Comment by dbwiddis** [comment]
Also you can type `diskutil list` to see them.  Here's mine... note `disk0` is the real, actual physical disk with the EFI partition, and gets read/write statistics while `disk1` is a virtual/synthesized disk which you can see from inside the OS as your main disk.  With "Fusion Drive" that synthesized disk could actually combine multiple other drives.
```
$ diskutil list
/dev/disk0 (internal, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      GUID_partition_scheme                        *1.0 TB     disk0
   1:                        EFI EFI                     314.6 MB   disk0s1
   2:                 Apple_APFS Container disk1         1.0 TB     disk0s2

/dev/disk1 (synthesized):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      APFS Container Scheme -                      +1.0 TB     disk1
                                 Physical Store disk0s2
   1:                APFS Volume Macintosh HD - Data     520.4 GB   disk1s1
   2:                APFS Volume Preboot                 1.8 GB     disk1s2
   3:                APFS Volume Recovery                1.1 GB     disk1s3
   4:                APFS Volume VM                      6.4 GB     disk1s4
   5:                APFS Volume Macintosh HD            8.9 GB     disk1s5
   6:              APFS Snapshot com.apple.os.update-... 8.9 GB     disk1s5s1
```
And here's an SO post I made several years ago with an earlier mac and slightly different output, but the same point: https://stackoverflow.com/questions/39221437/programmatically-get-osx-fusion-drive-information

**Comment by fireinrain** [comment]
> Also you can type `diskutil list` to see them. Here's mine... note `disk0` is the real, actual physical disk with the EFI partition, and gets read/write statistics while `disk1` is a virtual/synthesized disk which you can see from inside the OS as your main disk. With "Fusion Drive" that synthesized disk could actually combine multiple other drives.
> 
> ```
> $ diskutil list
> /dev/disk0 (internal, physical):
>    #:                       TYPE NAME                    SIZE       IDENTIFIER
>    0:      GUID_partition_scheme                        *1.0 TB     disk0
>    1:                        EFI EFI                     314.6 MB   disk0s1
>    2:                 Apple_APFS Container disk1         1.0 TB     disk0s2
> 
> /dev/disk1 (synthesized):
>    #:                       TYPE NAME                    SIZE       IDENTIFIER
>    0:      APFS Container Scheme -                      +1.0 TB     disk1
>                                  Physical Store disk0s2
>    1:                APFS Volume Macintosh HD - Data     520.4 GB   disk1s1
>    2:                APFS Volume Preboot                 1.8 GB     disk1s2
>    3:                APFS Volume Recovery                1.1 GB     disk1s3
>    4:                APFS Volume VM                      6.4 GB     disk1s4
>    5:                APFS Volume Macintosh HD            8.9 GB     disk1s5
>    6:              APFS Snapshot com.apple.os.update-... 8.9 GB     disk1s5s1
> ```
> 
> And here's an SO post I made several years ago with an earlier mac and slightly different output, but the same point: https://stackoverflow.com/questions/39221437/programmatically-get-osx-fusion-drive-information

I get your mean. But another question, how can i know which disk is real or which is virtual?
Is there any tag I can get from this lib?

**Comment by dbwiddis** [comment]
For now it seems reasonable to search for a partition named EFI, and a second partition with exactly the same file size as the next incremented numbered disk.

For fusion drives or FileVault, OSHI parses the `diskutil cs list` output for those.

It looks like it could be possible to parse `diskutil apfs list` to connect this setup, showing that disk0s2 is the same as disk 1.  Not sure how to expose that information to users.... any suggestions?
```
➜  diskutil apfs list
APFS Container (1 found)
|
+-- Container disk1 B7877CDC-8218-434B-BF18-E78A2E3C325F
    ====================================================
    APFS Container Reference:     disk1
    Size (Capacity Ceiling):      499963174912 B (500.0 GB)
    Capacity In Use By Volumes:   107683680256 B (107.7 GB) (21.5% used)
    Capacity Not Allocated:       392279494656 B (392.3 GB) (78.5% free)
    |
    +-< Physical Store disk0s2 3DD49631-094C-465D-AEBC-D5C733D652DA
    |   -----------------------------------------------------------
    |   APFS Physical Store Disk:   disk0s2
    |   Size:                       499963174912 B (500.0 GB)
    |
    +-> Volume disk1s1 09E4F9AB-F213-490B-A6C5-F16BEE2D6735
    |   ---------------------------------------------------
    |   APFS Volume Disk (Role):   disk1s1 (System)
    |   Name:                      Macintosh HD (Case-insensitive)
    |   Mount Point:               Not Mounted
    |   Capacity Consumed:         15414902784 B (15.4 GB)
    |   Sealed:                    Yes
    |   FileVault:                 Yes (Unlocked)
    |   Encrypted:                 No
    |   |
    |   Snapshot:                  3BF7CA93-443C-44C5-A82F-BECF6BF6D744
    |   Snapshot Disk:             disk1s1s1
    |   Snapshot Mount Point:      /
    |   Snapshot Sealed:           Yes
    |
```

**Comment by dbwiddis** [comment]
I know you closed this issue, but I just found something else that may be really helpful here.

I just got a new mac, and of course the first thing I did was open OSHI on it to test things. :-)

I have not just two, but four apparent disks even though I have only one internal (but I turned on filevault).  However, I noticed in OSHI's output the "major" number was the same on all of them.  That may be a way for you to identify unique hardware.
